internal/kms/kes.go

```diff
@@ -138,9 +140,44 @@ func NewWithConfig(config Config) (KMS, error) {
 			}
 		}
 	}()
+
+	go c.refreshKMSMasterKeyCache(kmsLogger)
 	return c, nil
 }
 
+// Request KES keep an up-to-date copy of the KMS master key to allow minio to start up even if KMS is down. The
+// cached key may still be evicted if the period of this function is longer than that of KES .cache.expiry.unused
+func (c *kesClient) refreshKMSMasterKeyCache(logger Logger) {
+	ctx := context.Background()
+
+	defaultCacheInterval := 10
+	cacheInterval, err := env.GetInt("EnvKESKeyCacheInterval", defaultCacheInterval)
+	if err != nil {
+		cacheInterval = defaultCacheInterval
+	}
+
+	timer := time.NewTimer(time.Duration(cacheInterval) * time.Second)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-timer.C:
+			var wg sync.WaitGroup
```

Comment:
Why this wait group? Not very useful you can keep it blocking just like before.

I mean wait group here is blocking as well when you have only 1 goroutine

Reply:
True. I've implemented your suggestion.